### ISSUE TITLE
TC001/TC002/TC003 false negative when using a type as generic parameter bound

### ISSUE THREAD

**Issue by flying-sheep** [astral-sh/ruff#27369]
### Summary

`TC001`/`TC002`/`TC003` don’t identify an import as type-only if it’s used as a bound in a PEP 695 type-parameter list (`def f[T: X](...)`).

[Repro](https://play.ruff.rs/1094cb13-8254-4b77-bd6c-280c0eba7ca2)

```python
from __future__ import annotations

from .anndata import AnnData

def f[T: AnnData](a: T) -> T: ...  # no violation flagged here
```

This is of course flagged correctly:

```python
from __future__ import annotations

from .anndata import AnnData

def f(a: AnnData) -> None: ...  # TC001 Move application import `.anndata.AnnData` into a type-checking block
```

### Version

ruff 0.16.0

**Comment by ntBre** [comment]
I think this is related to #16981 and https://github.com/astral-sh/ruff/issues/16886.

**Comment by flying-sheep** [comment]
Not really? Maybe the internal code is, but these issues are about `type` statements, this is about generic bounds.

**Comment by ntBre** [comment]
I should have linked to this comment: https://github.com/astral-sh/ruff/pull/16981#issuecomment-2766512370, which points out that PEP-695 generics are not included in the PR but involve the same design decisions.
